### ISSUE TITLE
Question about 'time' crate dependency version

### ISSUE THREAD

**Issue by h7kanna** [progval/unicode_names2#36]
Hi,

`unicode_names2_generator` crate's `time` dependency is pinned to be `<0.3.21`
Any reason why this is done and can it be lifted?

This is giving dep resolution issues when using with crates that depend on recent 'time' versions
Example

```
error: failed to select a version for `time`.
    ... required by package `gix-date v0.8.1`
    ... which satisfies dependency `gix-date = "^0.8.1"` of package `gix v0.56.0`
    ... which satisfies dependency `gix = "^0.56.0"` of package `activity-git v0.1.0 (/Users/h7kanna/Repos/activity-git)`
versions that meet the requirements `^0.3.23` are: 0.3.30, 0.3.29, 0.3.28, 0.3.27, 0.3.26, 0.3.25, 0.3.24, 0.3.23

all possible versions conflict with previously selected packages.

  previously selected package `time v0.3.0`
    ... which satisfies dependency `time = "^0.3, <0.3.21"` of package `unicode_names2_generator v1.1.0`
    ... which satisfies dependency `unicode_names2_generator = "^1.1.0"` of package `unicode_names2 v1.1.0`
    ... which satisfies dependency `unicode_names2 = "^1.1.0"` of package
```

**Comment by progval** [comment]
>  Any reason why this is done

`time v0.3.21` removed support for the Rust version in Debian stable (1.63.0), which is the only source of binaries on the machine I use to upload crates.

> and can it be lifted?

I'd welcome an alternative way to make this crate compile on Rust 1.63.0, but fiddling with `cargo update` for hours didn't help.

**Comment by progval** [comment]
Resolved in v1.2.2 by making the dependency on time optional.
